esp/logging/loggingagent/cassandraloggingagent/CMakeLists.txt

```diff
@@ -39,22 +38,23 @@ include_directories (
      ${HPCC_SOURCE_DIR}/esp/bindings
      ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp
      ${HPCC_SOURCE_DIR}/esp/logging
-)
+  )
 
-ADD_DEFINITIONS( -D_USRDLL  -DMYSQLLOGGINGLISTENER_EXPORTS -DMYSQLLOGAGENT_EXPORTS )
+  ADD_DEFINITIONS( -D_USRDLL  -DMYSQLLOGGINGLISTENER_EXPORTS -DMYSQLLOGAGENT_EXPORTS )
 
-set ( SRCS
+  set ( SRCS
     cassandralogagent.cpp
-)
+  )
 
-HPCC_ADD_LIBRARY( cassandralogagent SHARED ${SRCS} )
+  HPCC_ADD_LIBRARY( cassandralogagent SHARED ${SRCS} )
 
-install ( TARGETS cassandralogagent RUNTIME DESTINATION bin LIBRARY DESTINATION lib )
-add_dependencies (cassandralogagent espscm)
-target_link_libraries ( cassandralogagent
-    cassandraembed
+  install ( TARGETS cassandralogagent RUNTIME DESTINATION bin LIBRARY DESTINATION lib )
+  add_dependencies (cassandralogagent espscm)
+  target_link_libraries ( cassandralogagent
     cassandra
+    $<TARGET_FILE:cassandraembed>
```

Comment:
@xwang2713 This is the change that I had to make to work around the linking issue.  Still not certain why I had to declare it as a path instead of just as the target.  I was thinking it had something to do with the soname but after looking at other so's we make, I'm not so certain.  If you could take a look at this, test it out, and if you have any ideas, please let me know.  The relevant changes will show up under the link.txt file that gets generated in the build directory for cassandralogagent.